src/components/structures/MessagePanel.js

```diff
@@ -477,7 +480,8 @@ module.exports = React.createClass({
             // here.
             return !this.props.suppressFirstDateSeparator;
         }
-        return Math.floor(prevEvent.getTs() / MILLIS_IN_DAY) !== Math.floor(nextEventTs / MILLIS_IN_DAY);
+        return Math.floor((prevEvent.getTs() - MILLIS_TZ_OFFSET) / MILLIS_IN_DAY) !==
+                Math.floor((nextEventTs - MILLIS_TZ_OFFSET) / MILLIS_IN_DAY);
```

Comment:
This looks like a rabbithole to me. It will fuck up at DST boundaries, or if a user changes the timezone settings on their computer.

How about just making two Date objects, an comparing the year, month, and date?